Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,124 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
-
-static FILE* get_stream(char const* filepath, char const* perms)
+static Optional<String> get_line_content(StringView line, size_t count, bool show_duplicates, bool print_count)
 {
-    FILE* ret;
-
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            return String::formatted("{} {}\n", count + 1, line);
+        else
+            return String::formatted("{}\n", line);
     }
+    return {};
+}
 
-    ret = fopen(filepath, perms);
-    if (ret == nullptr) {
-        perror("fopen");
-        exit(1);
+static StringView skip(StringView line, unsigned int char_skip_count, unsigned int field_skip_count)
+{
+    line = line.trim("\n"sv);
+    if (field_skip_count) {
+        bool in_field = false;
+        int field_index = 0;
+        unsigned int current_field = 0;
+        for (size_t i = 0; i < line.length(); i++) {
+            char c = line[i];
+            if (c == ' ' || c == '\t' || c == '\v' || c == '\f' || c == '\r')
+                in_field = false;
```

Comment:
You should add `{}` braces around this `if` since the else branch has them (otherwise it looks a bit odd).